### DISCUSSION TITLE
Make properties inside the existing object schema nullable

### DISCUSSION THREAD

**Discussion by ReaganM02** [colinhacks/zod#3283] [Q&A] [answered]
E.g. schema
```
const user = z.object({
  firstName: z.string(),
  lastName: z.string()
})
```

I am using this schema somewhere in other function so I cannot add nullable method inside this object. But this time,  I need to use this schema but the properties should be nullable. How do I achieve that without recreating the schema? 

The expected result is like this:
```
{
 firstName: string | null
 lastName: string | null
}
```
I don't want to use partial since it will make the property optional.

**Comment by JacobWeisenburger** [comment, marked as answer]
Is this what you are looking for?

```ts
const user = z.object( {
    firstName: z.string(),
    lastName: z.string()
} )

const nullableUser = user.extend( {
    firstName: user.shape.firstName.nullable(),
    lastName: user.shape.lastName.nullable()
} )

type NullableUser = z.output<typeof nullableUser>
// type NullableUser = {
//     firstName: string | null
//     lastName: string | null
// }
```

If you found my answer satisfactory, please consider supporting me. Even a small amount is greatly appreciated. Thanks friend! 🙏
https://github.com/sponsors/JacobWeisenburger

More info:
https://github.com/colinhacks/zod/discussions/863

**Reply by ReaganM02** [reply]
Thank you so much!
